From f2181fd41541c5b60263da36892c5cbf8bec4b49 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Fri, 23 Mar 2018 07:46:48 -0500
Subject: [PATCH 3537/5242] MLK-17703-4: drm: imx: dcss: return the hdr10
 table at once

commit  610d1960ac4af6767fc1798586ffd059bafe05ab from
https://source.codeaurora.org/external/imx/linux-imx.git

Don't go through the rest of the list if we found our table. Just return it
immediately.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-hdr10.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/imx/dcss/dcss-hdr10.c b/drivers/gpu/imx/dcss/dcss-hdr10.c
index 18a6615..aa4b1e2 100644
--- a/drivers/gpu/imx/dcss/dcss-hdr10.c
+++ b/drivers/gpu/imx/dcss/dcss-hdr10.c
@@ -357,16 +357,15 @@ static u32 *dcss_hdr10_find_tbl(u64 desc, struct list_head *head)
 {
 	struct list_head *node;
 	struct dcss_hdr10_tbl_node *tbl_node;
-	u32 *tbl = NULL;
 
 	list_for_each(node, head) {
 		tbl_node = container_of(node, struct dcss_hdr10_tbl_node, node);
 
 		if ((tbl_node->tbl_descriptor & desc) == desc)
-			tbl = tbl_node->tbl_data;
+			return tbl_node->tbl_data;
 	}
 
-	return tbl;
+	return NULL;
 }
 
 static int dcss_hdr10_get_tbls(struct dcss_hdr10_priv *hdr10, bool input,
-- 
1.7.9.5

